osrf/vrx
# WAM-V joystick teleoperation

Requires:
  * https://github.com/gazebosim/gz-sim/pull/1681

This pull request adds the ability to teleoperate the WAM-V, which I think it's very useful for testing scoring plugins.

As a reminder, you need a Logitech F310 controller with the back switch in `D` mode.

### How to test it?

Launch the simulation:
```
ros2 launch vrx_ros competition_local.launch.py ign_args:="-v 4 -r sydney_regatta.sdf"
```

Launch the WAM-V:
```
ros2 launch vrx_ign spawn.launch.py name:=wamv world:=sydney_regatta model:=wam-v x:=-532 y:=162 z:=0 R:=0 P:=0 Y:=1
```

Launch the `usv_joy_teleop` launch file:
```
ros2 launch vrx_ign usv_joy_teleop.py
```

To control the WAM-V you have to keep the `L1` button pressed (deadman button). Other than that, the controls are the same that we used to have. The ball shooter is not installed in the WAM-V but you can confirm that the message is sent by echoing the ` /wamv/shooters/ball_shooter/fire` topic while you pressed the `A` button.
```
ros2 topic echo /wamv/shooters/ball_shooter/fire
data: true
---
```

**Note**: I think that the [joy_teleop package](https://github.com/ros-teleop/teleop_tools/tree/foxy-devel/joy_teleop) is still not polished enough to be used with ROS2. As far as I know it's not possible to configure it without a deadman button, which I think it's a bug.
